src/bcs/VacuumConcBC.C

```diff
@@ -7,22 +7,37 @@ VacuumConcBC::validParams()
 {
   InputParameters params = IntegratedBC::validParams();
   params += ScalarTransportBase::validParams();
+  MooseEnum vacuum_bc_type("marshak mark milne", "marshak");
+  params.addParam<MooseEnum>("vacuum_bc_type", vacuum_bc_type,
+      "Whether to apply Marshak, Mark, or Milne vacuum boundary conditions. Defaults to Marshak.");
   return params;
 }
 
 VacuumConcBC::VacuumConcBC(const InputParameters & parameters)
   : IntegratedBC(parameters), ScalarTransportBase(parameters)
 {
+  switch (getParam<MooseEnum>("vacuum_bc_type"))
+  {
+    case MARSHAK:
+      _alpha = 2.;
+      break;
+    case MARK:
+      _alpha = std::sqrt(3.);
+      break;
+    case MILNE:
+      _alpha = 3 * 0.710446;
```

Comment:
This random number feels random. What is 0.710446 and should there be an explanation somewhere?

Reply:
It's the coefficient for the extrapolation length in the Milne problem (1D one-group half plane problem) $d=0.710446\lambda_{tr}=0.710446\times3D$. I'll include a brief in-line comment here and a longer description in the documentation file.

Reply:
if you can assign a variable name to this number, then, that would be good.